projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -235,11 +236,18 @@ export class MultiSelectComponent<V> implements ControlValueAccessor, AfterConte
 
   public ngAfterContentInit(): void {
     this.allOptions$ = this.allOptionsList !== undefined ? queryListAndChanges$(this.allOptionsList) : EMPTY;
-    this.filteredOptions$ = combineLatest([this.allOptions$, this.caseInsensitiveSearchSubject]).pipe(
+    this.filteredOptions$ = combineLatest([
+      this.allOptions$,
+      this.caseInsensitiveSearchSubject,
+      this.reorderSelectedItemsSubject
+    ]).pipe(
       map(([options, searchText]) =>
         isEmpty(searchText)
-          ? options.toArray()
-          : options.filter(option => option.label.toLowerCase().includes(searchText.toLowerCase()))
+          ? this.reorderSelectedItemsFirst(options.toArray(), this.selected)
+          : this.reorderSelectedItemsFirst(
+              options.filter(option => option.label.toLowerCase().includes(searchText.toLowerCase())),
+              this.selected
+            )
```

Comment:
we don't need to worry about reordering every time. reordering needs to happen on two occasions
1. On init.
2. after the user closes the popover, reorder in the background.

We shouldn't need to worry about reordering every time there's a search.

The way to achieve that would be to not combine reorder subject and search subject. do the reordering before the search.

Reply:
done